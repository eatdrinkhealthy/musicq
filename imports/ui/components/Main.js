// @flow
import React from "react";
import Header from "./Header";

type IProps = {
  children: React.Element<*>,
};

const Main = (props: IProps): React.Element<*> => (
  <div>
    <Header />
    { React.cloneElement(props.children) }
  </div>
);

export default Main;

