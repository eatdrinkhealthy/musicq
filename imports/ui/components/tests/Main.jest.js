// TODO   add flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import Main from "../Main";

describe("<Main />", function () {
  it("matches render snaphot", function () {
    const MainWithChildElement = (<Main><div>some text</div></Main>);

    const tree = renderer.create(MainWithChildElement).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

