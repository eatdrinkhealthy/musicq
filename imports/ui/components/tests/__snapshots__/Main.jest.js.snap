exports[`<Main /> matches render snaphot 1`] = `
<div>
  <div>
    <h1>
      MusicQ
    </h1>
  </div>
  <div>
    some text
  </div>
</div>
`;
